pkg/reconciler/messaging/channel/channel.go

```diff
@@ -286,22 +300,12 @@ func (r *Reconciler) syncSubscribersStatus(ctx context.Context, channel *v1beta1
 
 func (r *Reconciler) reconcileTopic(ctx context.Context, channel *v1beta1.Channel) (*inteventsv1beta1.Topic, error) {
 	topic, err := r.getTopic(ctx, channel)
-	if err != nil && !apierrors.IsNotFound(err) {
-		logging.FromContext(ctx).Desugar().Error("Unable to get a Topic", zap.Error(err))
-		return nil, err
-	}
-	if topic != nil {
-		if topic.Status.Address != nil {
-			channel.Status.SetAddress(topic.Status.Address.URL)
-		} else {
-			channel.Status.SetAddress(nil)
-		}
-		return topic, nil
-	}
+
```

Comment:
It feels odd for this to be separated from the `if err != nil` line, can it be moved just before that one?

Reply:
which one do you mean? the `if topic != nil` stuff?

Reply:
I mean that we set `err` here on line 302, but don't check it until line 317.

Reply:
got it, indeed, it does make sense. Moved it